### PR TITLE
Replace wall/indestructible/destructible typepath with mapping helper

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

#78239 is a fun mapping add but engineered in a way that 
1. Creates a real eyesore of a typepath
2. Would further proliferate a hundred subtypes if it became commonly used

Instead of using subtypes for this I put the behaviour in a component and made a mapping helper to apply the component.
Now you can just put the mapping helper on top of any turf you want to make into a zelda bomb wall and it will be so, rather than having to make different subtypes for walls with different icons.

## Why It's Good For The Game

Cleaner, more maintainable.

## Changelog

not player facing

### CODE REVIEW

File: code/datums/components/bombable_turf.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Apply this to a turf (usually a wall) and it will be destroyed instantly by any explosion.
+ * Most walls can already be destroyed by explosions so this is largely for usually indestructible ones.
+ * For applying it in a map editor, use /obj/effect/mapping_helpers/bombable_wall
+ */
+/datum/component/bombable_turf
+	/// Overlay we show to let you know it can be blown up
+	var/mutable_appearance/overlay
```
Comment (Ghommie) [inline code/datums/components/bombable_turf.dm:8]: If the overlays could be added during as a signal proc for whatever signal update_overlays sends, then we could easily make this into an element instead.

**Comment by san7890** [comment]
can you make an updatepaths for this? i don't think it's proliferated as much in mapping yet but it's nice for people who've already started cracking away at adding this

**Comment by Jacquerel** [comment]
> can you make an updatepaths for this? i don't think it's proliferated as much in mapping yet but it's nice for people who've already started cracking away at adding this

i mean i can but it was merged 4 hours before i opened this pr so i feel like we'll _probably_ squeak by

File: tools/UpdatePaths/Scripts/78365_explodable_walls.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+/turf/closed/indestructible/explodable/@SUBTYPES : /turf/closed/indestructible/@SUBTYPES{@OLD}, /obj/effect/mapping_helpers/bombable_wall
```
Comment (san7890) [inline tools/UpdatePaths/Scripts/78365_explodable_walls.txt:1]:
```suggestion
/turf/closed/indestructible/explodable/@SUBTYPES : /turf/closed/indestructible/@SUBTYPES, /obj/effect/mapping_helpers/bombable_wall
```

turf varedits deserve to die

**Review by Ghommie — APPROVED** [review]

**Pull request merged**